[8.0][html_text] Excerpt generator.

yajo opened the pull request:
This module adds a technical utility to allow the developer to get an excerpt from any HTML chunk.

You can choose the maximum amount of words or characters if you want.

See the README and inline docstrings for further details. Example usage found in https://github.com/OCA/website/pull/163.

Commit:
[8.0][html_text] Excerpt generator.

This module adds a technical utility to allow the developer to get an excerpt from any HTML chunk.

You can choose the maximum amount of words or characters if you want.

See the README and inline docstrings for further details.

elicoidal commented on html_text/README.rst:
```diff
@@ -0,0 +1,81 @@
+Text from HTML field
+====================
+
+This module extracts text from any chunk of HTML, without HTML tags or
```
This module extracts text from any chunk of HTML, without HTML tags or
attributes. You can chose either:
- to truncate the result by amount of words or characters.
- to append an ellipsis (or any character(s)) at the end of the result.

yajo replied:
Fixed.

pedrobaeza replied:
I think it's better to put "This module provides some technical modules that allow to extract ..."

elicoidal commented on html_text/models/ir_fields_converter.py:
```diff
@@ -0,0 +1,55 @@
+# © 2016 Grupo ESOC Ingeniería de Servicios, S.L.U. - Jairo Llopis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from lxml import html
```
Should you protect the import (Try...Except)?

yajo replied:
This is an Odoo's dependency, so it is supposed to be always installed.

Commit:
Better phrasing.

Commit:
Avoid failures when wrong HTML is recieved.

Otherwise QWeb views would get broken badly.

Commit:
100% coverage.

Add a new `fail` parameter to allow exceptions to bubble up.

Control `ParserError` exceptions too.

yajo commented:
Could somebody please tag this as "needs review"?

pedrobaeza commented:
:+1:

pedrobaeza commented:
Tagged as needs review

pedrobaeza commented:
Little comment, but :+1:

atchuthan commented:
code reviewed :+1:

yajo committed:
Better explanation.

yajo commented:
Fixed.
